app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -4,8 +4,36 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
   require_authentication :all, scopes: [:project]
   schema_type :json_schema
 
+  def update_project_settings
+    binding.pry
+    project = Project.find params[:project_id]
+    unless required_params
+      render json: { error: 'missing or invalid params' }, status: :unprocessable_entity
+    end
+    if project.owner?(api_user.user)
+      upp = UserProjectPreference.find_or_initialize_by(project: project, user_id: api_user.user)
+      upp.settings.merge! params[:settings]
+      if upp.save!
+        render status: :ok, nothing: true
+      else
+        render status: :unprocessable_entity, nothing: true
+      end
+    else
+      render json: { error: 'not authorized' }, status: :not_authorized
+    end
+  end
+
   private
 
+  def required_params
+    params_to_check = [ params[:project_id], params[:user_id], params[:settings] ]
+    params_to_check.all? { |param| !param.blank? }
+  rescue JsonSchema::ValidationError,
```

Comment:
these will be taken care of normally via the [api_controller](https://github.com/zooniverse/Panoptes/blob/a89ff672762f5b403f85c1c007f18db134bf62f0/app/controllers/api/api_controller.rb#L14)